common/src/main/java/org/apache/falcon/entity/parser/FeedEntityParser.java

```diff
@@ -157,14 +157,30 @@ private void validateLifecycle(Feed feed) throws FalconException {
         LifecyclePolicyMap map = LifecyclePolicyMap.get();
         for (Cluster cluster : feed.getClusters().getClusters()) {
             if (FeedHelper.isLifecycleEnabled(feed, cluster.getName())) {
-                if (FeedHelper.getRetentionStage(feed, cluster.getName()) == null) {
-                    throw new ValidationException("Retention is a mandatory stage, didn't find it for cluster: "
+                if ((FeedHelper.getRetentionStage(feed, cluster.getName()) != null) || (FeedHelper.getArchivalStage(feed, cluster.getName()) != null)) {
+                    validateRetentionStage(feed, cluster);
+                    validateArchivalStage(feed, cluster);
+                    for (String policyName : FeedHelper.getPolicies(feed, cluster.getName())) {
```

Comment:
Once you split the getPolicies method. Use the appropriate ones inside validateRetentionStage and validateArchivalStage.